### PR TITLE
Can we export exceptions?

### PR DESCRIPTION
**Pull request by kngwyu**
In this PR I added a new macro named `export_exceptions`, which inserts
```rust
#[pymethods]
fn _pyo3_exceptions(_py: pyo3::Python, m: &pyo3::types::PyModule) -> pyo3::PyResult<()> {
    use pyo3::type_object::PyTypeObject;
    m.add("PyBorrowError", pyo3::pycell::PyBorrowError::type_object())?;
    m.add(
        "PyBorrowMutError",
        pyo3::pycell::PyBorrowMutError::type_object(),
    )?;
    Ok(())
}
```
to exension, so that our custom exceptions can be imported to Python code.

But somehow `except PyBorrowError` fails oddly like
```python
____________________________________________________________ test_catch_borrrow ____________________________________________________________

    def test_catch_borrrow():
        m = Mutable()
        try:
            m.invalid_borrow(m)
            assert False
        except PyBorrowError:
            pass
        except Exception as e:
>           raise e

tests/test_pycell_exceptions.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def test_catch_borrrow():
        m = Mutable()
        try:
>           m.invalid_borrow(m)
E           _pyo3_exceptions.PyBorrowError
```
and also `isintance(e, PyBorrowError)` returns False.

### CODE REVIEW

File: tests/test_module.rs [hunk 1]
```diff
@@ -276,3 +276,38 @@ fn test_vararg_module() {
...
+#[pymodule]
+fn module_with_exceptions(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<ValueClass>()?;
+    m.add_wrapped(pyo3::wrap_pyfunction!(invalid_mutation))?;
+    m.add_wrapped(pyo3::wrap_pymodule!(_pyo3_exceptions))?;
+    Ok(())
+}
+
+#[test]
+fn catch_custom_exceptions() {
```
Comment (kngwyu) [inline tests/test_module.rs]: This test succeeds.

File: examples/rustapi_module/tests/test_pycell_exceptions.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from rustapi_module.cell import Mutable
```
Comment (kngwyu) [inline examples/rustapi_module/tests/test_pycell_exceptions.py]: These tests fail.

**Comment by birkenfeld** [comment]
Maybe it's easier to define the module in Python (some global code running once when PyO3 initializes), and use the exception class from there like other Python exceptions?

**Comment by kngwyu** [comment]
Closed since it's outdated now, though exceptions should eventually get able to be exported

**Pull request closed without merging**